### PR TITLE
Ticket service func

### PR DESCRIPTION
**Pull request by OlgaKochneva**
Add partial ticket-service functionality

### CODE REVIEW

**Comment by OlgaKochneva** [comment]
[[BE] Add partial functionality to ticket](https://trello.com/c/rJgCXHuJ/94-be-add-partial-functionality-to-ticket)

File: src/main/java/com/epam/lowcost/controller/TicketController.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package com.epam.lowcost.controller;
+
+import com.epam.lowcost.services.interfaces.TicketService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.stereotype.Controller;
+
+
+@Controller
+@PreAuthorize("hasAuthority('ROLE_ADMIN')")
```
Comment (vakhonin) [inline src/main/java/com/epam/lowcost/controller/TicketController.java]: why only admin ?

File: src/main/java/com/epam/lowcost/services/implementations/TicketServiceImpl.java [hunk 2]
```diff
@@ -20,26 +25,68 @@ public TicketServiceImpl(TicketRepository ticketRepository) {
...
-        return null;
+    public long numberBoughtPlacesForEachClass(long flightId, boolean isBusiness) {
+        return ticketRepository.countAllByFlight_IdAndIsBusinessAndIsDeleted(flightId, isBusiness, false);
+    }
+
+    @Override
+    public String deleteTicketsByFlightId(long flightId) {
+        List<Ticket> ticketsToDelete = getAllTicketsForCurrentFlight(flightId);
+        ticketsToDelete.forEach(f -> deleteTicketById(f.getId()));
+        return "All tickets for flight " + flightId + "deleted";
```
Comment (vakhonin) [inline src/main/java/com/epam/lowcost/services/implementations/TicketServiceImpl.java]: why hardcode ?

File: src/main/java/com/epam/lowcost/controller/TicketController.java (hunk 1 above)
Comment (OlgaKochneva) [reply src/main/java/com/epam/lowcost/controller/TicketController.java]: delete this

File: src/main/java/com/epam/lowcost/services/implementations/TicketServiceImpl.java (hunk 2 above)
Comment (OlgaKochneva) [reply src/main/java/com/epam/lowcost/services/implementations/TicketServiceImpl.java]: done

File: src/main/java/com/epam/lowcost/services/interfaces/TicketService.java [hunk 3]
```diff
@@ -1,19 +1,22 @@
...
 
     Ticket addTicket(Ticket ticket);
 
-    Ticket updateTicket(Ticket ticket);
+    String deleteTicketById(long id);
 
-    String deleteTicket(long id);
+    long getNumberBoughtPlacesForEachClass(long flightId, boolean isBusiness);
 
+    @Transactional
```
Comment (vakhonin) [inline src/main/java/com/epam/lowcost/services/interfaces/TicketService.java]: in Services

File: src/main/java/com/epam/lowcost/controller/TicketController.java [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+import com.epam.lowcost.services.interfaces.TicketService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+
+
+@Controller
+public class TicketController {
+    @Autowired
+    TicketService ticketService;
```
[the remark this answers is not in the document]
Comment (OlgaKochneva) [reply src/main/java/com/epam/lowcost/controller/TicketController.java]: now private and final

**Pull request merged**